### PR TITLE
Improve error message for grad-requiring tensor to numpy conversion

### PR DESCRIPTION
**Pull request by unacanal**
The suggestion to use `tensor.detach().numpy()` may not be valid when the tensor is on a device other than CPU.

Moving the tensor to CPU will ensure that the suggestion will be always valid.

### CODE REVIEW

File: torch/csrc/utils/tensor_numpy.cpp [hunk 1]
```diff
@@ -145,7 +145,7 @@ PyObject* tensor_to_numpy(const at::Tensor& tensor, bool force /*=false*/) {
     TORCH_CHECK(
         !(at::GradMode::is_enabled() && tensor.requires_grad()),
         "Can't call numpy() on Tensor that requires grad. "
-        "Use tensor.detach().numpy() instead.");
+        "Use tensor.detach().cpu().numpy() instead.");
```
Comment (colesbury) [inline torch/csrc/utils/tensor_numpy.cpp:148]: @ezyang - I think this is getting silly. `tensor.numpy()` should just give you the numpy array. If it needs to make a copy, such so be it.

The lesson I took from `.reshape()` and `.view()` is that APIs that just do the desired operation (`reshape()`) are a much better experience than erroring out if the operation requires a copy. The guaranteed aliasing that `.view()` and `.numpy()` is not important enough to make the general experience worse.

Comment (ezyang) [reply torch/csrc/utils/tensor_numpy.cpp:148]: Well, the fortunate thing here is, because we are raising an error, it is easy to make it implicitly convert now without breaking BC :)

Comment (albanD) [reply torch/csrc/utils/tensor_numpy.cpp:148]: I'm not sure about this change, this error is specifically about the requires_grad part. If you want an all encompassing fix here, i would suggest `tensor.numpy(force=True)` instead of the partial version here
